File: VB/T424872/CustomWebDocumentViewerReportResolver.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports DevExpress.XtraReports.Web.WebDocumentViewer
Imports DevExpress.XtraReports.Web.Extensions
Imports System.IO
Imports DevExpress.XtraReports.UI

Namespace T424872
	Public Class CustomWebDocumentViewerReportResolver
		Implements IWebDocumentViewerReportResolver

		Private privateReportStorage As ReportStorageWebExtension
		Public Property ReportStorage() As ReportStorageWebExtension
			Get
				Return privateReportStorage
			End Get
			Private Set(ByVal value As ReportStorageWebExtension)
				privateReportStorage = value
			End Set
		End Property

		Public Sub New()
			Me.ReportStorage = New CustomReportStorageWebExtension()
		End Sub

		Public Function Resolve(ByVal reportName As String) As DevExpress.XtraReports.UI.XtraReport Implements IWebDocumentViewerReportResolver.Resolve
			Dim report As New XtraReport()

			Dim reportLayout() As Byte = ReportStorage.GetData(reportName)
			Using ms As New MemoryStream(reportLayout)
				report.LoadLayout(ms)
			End Using
			Return report
		End Function
	End Class
End Namespace

File: CS/T424872/CustomWebDocumentViewerReportResolver.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using DevExpress.XtraReports.Web.WebDocumentViewer;
using DevExpress.XtraReports.Web.Extensions;
using System.IO;
using DevExpress.XtraReports.UI;

namespace T424872 {
    public class CustomWebDocumentViewerReportResolver : IWebDocumentViewerReportResolver {
        public ReportStorageWebExtension ReportStorage { get; private set; }

        public CustomWebDocumentViewerReportResolver() {
            this.ReportStorage = new CustomReportStorageWebExtension();
        }

        public DevExpress.XtraReports.UI.XtraReport Resolve(string reportName) {
            XtraReport report = new XtraReport();

            byte[] reportLayout = ReportStorage.GetData(reportName);
            using (MemoryStream ms = new MemoryStream(reportLayout)) {
                report.LoadLayout(ms);    
            }
            return report;
        }
    }
}